### PR TITLE
Un-deprecate `Plumber$run(swaggerCallback=, debug=)` and add `Plumber$run(docs=,quiet=)`

### PR DESCRIPTION
**Pull request by jcheng5**
To better support headless running scenarios. Specifically, I'm creating [unit tests for plumbertableau](https://github.com/rstudio/plumbertableau/pull/8) that do not involve using an additional process.

## Minimal reproducible example

```r
# Messages printed, swagger UI launched
pr() %>% pr_run()

# No messages printed, no swagger UI launched
pr() %>% pr_run(swaggerCallback = NULL, quiet = TRUE)
```
PR task list:
- [x] Update NEWS
- [x] Add tests
- [x] Update documentation with `devtools::document()`

### CODE REVIEW

**Comment by meztez** [comment]
This would probably work too
```r
pr() %>% pr_set_docs_callback(NULL) %>% pr_run(quiet = TRUE)
```

It made realize `options_plumber` needs to handle deprecated option `plumber.swagger.url` so we could do

```r
options_plumber(docs.callback = NULL) # Whole session
pr() %>% pr_run(quiet = TRUE)
```

As it is we still default to it when initializing the router.

I liked the idea of distanciating the package from the swagger ui by using more general names.

**Comment by schloerke** [comment]
@jcheng5 `@meztez` has a fix for the `docs.callback` option for a global session. (PR fix: https://github.com/rstudio/plumber/pull/766). I'd prefer to not un-deprecate `pr_run(swaggerCallback=)` and use `pr_set_docs_callback(NULL)` instead.

--------------------------

Would Connect ever want access to `pr_run(quiet=)`? Do you think we should allow users to ever receive the httpuv server to do as they please?

If no, then it seems like an appropriate place to put the `quiet=` argument (and end of the PR). If yes, then we should prolly make a `pr_set_httpuv(quiet = FALSE, sync = TRUE)` like method to do this before the `pr_run()` call.
* `quiet` to turn off printing
* `method` to switch between `httpuv::runServer()`, `httpuv::startServer()`, or return the server before starting. This parameter will take more implementation thought to make sure all existing assumptions (such as any `on.exit()` call) are covered.

While we do not need to implement the `method` parameter. I'd like to put the `quiet` parameter in an appropriate spot with room to grow.

**Comment by schloerke** [comment]
Talking more with `@wch`, the httpuv methods and alterations would prolly be done best with a method outside of `pr_run()`.

Might be good to add a `pr_build()` which preps the plumber router and `pr_run()` would wrap `pr_build()` and then call `httpuv::runServer()`. For the concept of using different `httpuv` methods, it would require an update to `httpuv` such as adding a `app$onAppStop(fn)` (similar to `app$onWSClose()`). 

-------------

tl/dr; Put quiet in `pr_run(quiet=)` and `httpuv` related things could be added in new methods.

File: R/plumber.R [hunk 1]
```diff
@@ -145,14 +145,18 @@ Plumber <- R6Class(
     #' This value does not need to be explicitly assigned. To explicitly set it, see [options_plumber()].
     #' @param debug Deprecated. See `$setDebug()`
     #' @param swagger Deprecated. See `$setDocs(docs)` or `$setApiSpec()`
-    #' @param swaggerCallback Deprecated. See `$setDocsCallback()`
+    #' @param swaggerCallback An optional single-argument function that is
+    #'   called back with the URL to an OpenAPI user interface when one becomes
+    #'   ready. If missing, defaults to `$setDocsCallback()`.
```
Comment (schloerke) [inline R/plumber.R:150]:
```suggestion
    #' @param swaggerCallback Deprecated. See `$setDocsCallback()`
```

File: R/plumber.R [hunk 2]
```diff
@@ -145,14 +145,18 @@ Plumber <- R6Class(
...
+    #' @param quiet If `TRUE`, don't print routine startup messages.
     #' @importFrom lifecycle deprecated
     run = function(
       host = '127.0.0.1',
       port = getOption('plumber.port', NULL),
       swagger = deprecated(),
       debug = deprecated(),
-      swaggerCallback = deprecated()
+      swaggerCallback,
```
Comment (schloerke) [inline R/plumber.R:158]: Add in `...`
```suggestion
      swaggerCallback = deprecated(),
      ...,
```

File: R/plumber.R [hunk 3]
```diff
@@ -184,14 +188,16 @@ Plumber <- R6Class(
           }
         }
       }
-      if (lifecycle::is_present(swaggerCallback)) {
-        lifecycle::deprecate_warn("1.0.0", "run(swaggerCallback = )", "setDocsCallback(callback = )")
-        self$setDocsCallback(swaggerCallback)
+
+      if (missing(swaggerCallback)) {
+        swaggerCallback <- private$docs_callback
```
Comment (schloerke) [inline R/plumber.R:193]:
```suggestion
      if (lifecycle::is_present(swaggerCallback)) {
        lifecycle::deprecate_warn("1.0.0", "run(swaggerCallback = )", "setDocsCallback(callback = )")
        self$setDocsCallback(swaggerCallback)
```

File: R/plumber.R [hunk 4]
```diff
@@ -205,7 +211,8 @@ Plumber <- R6Class(
           host = host,
           port = port,
           docs_info = private$docs_info,
-          callback = private$docs_callback
+          callback = swaggerCallback,
```
Comment (schloerke) [inline R/plumber.R:214]:
```suggestion
          callback = private$docs_callback,
```

**Comment by meztez** [comment]
Running an headless plumber would make a lot of people happy.

File: R/pr.R [hunk 5]
```diff
@@ -475,6 +475,10 @@ pr_filter <- function(pr,
 #' @param port A number or integer that indicates the server port that should
 #' be listened on. Note that on most Unix-like systems including Linux and
 #' Mac OS X, port numbers smaller than 1025 require root privileges.
+#' @param swaggerCallback An optional single-argument function that is called
+#'   back with the URL to an OpenAPI user interface when one becomes ready. If
+#'   missing, defaults to `$setDocsCallback()`.
```
Comment (schloerke) [inline R/pr.R:480]:
```suggestion
```

File: R/plumber.R [hunk 6]
```diff
@@ -145,14 +145,18 @@ Plumber <- R6Class(
     #' This value does not need to be explicitly assigned. To explicitly set it, see [options_plumber()].
     #' @param debug Deprecated. See `$setDebug()`
     #' @param swagger Deprecated. See `$setDocs(docs)` or `$setApiSpec()`
-    #' @param swaggerCallback Deprecated. See `$setDocsCallback()`
+    #' @param swaggerCallback An optional single-argument function that is
+    #'   called back with the URL to an OpenAPI user interface when one becomes
+    #'   ready. If missing, defaults to `$setDocsCallback()`.
+    #' @param quiet If `TRUE`, don't print routine startup messages.
```
Comment (schloerke) [inline R/plumber.R:151]:
```suggestion
    #' @param ... Should be empty
    #' @param quiet If `TRUE`, don't print routine startup messages.
```

File: R/plumber.R [hunk 7]
```diff
@@ -145,14 +145,18 @@ Plumber <- R6Class(
...
       host = '127.0.0.1',
       port = getOption('plumber.port', NULL),
       swagger = deprecated(),
       debug = deprecated(),
-      swaggerCallback = deprecated()
+      swaggerCallback,
+      quiet = FALSE
     ) {
 
       if (isTRUE(private$disable_run)) {
```
Comment (schloerke) [inline R/plumber.R:162]: Make sure the dots are empty

```suggestion
      ellipsis::check_dots_empty()
      
      if (isTRUE(private$disable_run)) {
```

File: R/pr.R [hunk 8]
```diff
@@ -475,6 +475,10 @@ pr_filter <- function(pr,
...
+#' @param quiet If `TRUE`, don't print routine startup messages.
```
Comment (schloerke) [inline R/pr.R:481]:
```suggestion
#' @param ... Should be empty.
#' @param quiet If `TRUE`, don't print routine startup messages.
```

File: R/pr.R [hunk 9]
```diff
@@ -488,11 +492,15 @@ pr_filter <- function(pr,
 #' @export
 pr_run <- function(pr,
                    host = '127.0.0.1',
-                   port = getOption('plumber.port', NULL)
+                   port = getOption('plumber.port', NULL),
+                   swaggerCallback,
```
Comment (schloerke) [inline R/pr.R:496]:
```suggestion
                   ...,
```

File: R/pr.R [hunk 10]
```diff
@@ -488,11 +492,15 @@ pr_filter <- function(pr,
...
+                   quiet = FALSE
 ) {
   validate_pr(pr)
```
Comment (schloerke) [inline R/pr.R:499]:
```suggestion
  ellipsis::check_dots_empty()
  validate_pr(pr)
```

File: R/pr.R [hunk 11]
```diff
@@ -488,11 +492,15 @@ pr_filter <- function(pr,
...
   pr$run(host = host,
-         port = port)
+         port = port,
+         swaggerCallback = swaggerCallback,
```
Comment (schloerke) [inline R/pr.R:502]:
```suggestion
```

File: NEWS.md [hunk 12]
```diff
@@ -23,6 +23,8 @@ plumber 1.0.0.9999 Development version
 
 * OpenAPI Specification can be set using a file path. (@meztez #696)
 
+* Added `quiet = TRUE` to `Plumber$run()` to suppress routine startup messages. (@jcheng5 #765)
```
Comment (schloerke) [inline NEWS.md:26]:
```suggestion
* Added `quiet = TRUE` to `pr_run()` and `Plumber$run()` to suppress routine startup messages. (@jcheng5 #765)
```

**Review by schloerke — CHANGES_REQUESTED** [review]
I believe this should revert the swaggerCallback changes and also insert `...` so that `quiet` must be full named.

**Comment by schloerke** [comment]
I am going to resolve all requested changes above as they are not to be used.

File: NEWS.md [hunk 13]
```diff
@@ -23,6 +23,12 @@ plumber 1.0.0.9999 Development version
 
 * OpenAPI Specification can be set using a file path. (@meztez #696)
 
+* Un-deprecated `Plumber$run(debug=, swaggerCallback=)` and added the parameters for `Plumber$run(docs=, quiet=)` and `pr_run(debug=, docs=, swaggerCallback=, quiet=)`. Now, all four parameters will not produce lingering effects on the `Plumber` router. (@jcheng5 #765)
+  * Setting `quiet = TRUE` will suppress routine startup messages.
+  * Setting `debug = TRUE`, will display information when an error occurs. See `pr_set_debug()`.
+  * Setting `docs` will update the visual documentation. See `pr_set_docs()`.
+  * Setting `swaggerCallback` to a function which will be called with a url to the documentation. See `pr_set_docs_callback()`.
```
Comment (jcheng5) [inline NEWS.md:30]:
```suggestion
  * Set `swaggerCallback` to a function which will be called with a url to the documentation, or `NULL` to do nothing. See `pr_set_docs_callback()`.
```

File: R/pr.R [hunk 14]
```diff
@@ -475,24 +475,54 @@ pr_filter <- function(pr,
...
 #' \dontrun{
 #' pr() %>%
 #'   pr_run()
 #'
 #' pr() %>%
-#'   pr_run(port = 5762)
+#'   pr_run(
+#'     # manually set port
+#'     port = 5762,
+#'     # turn of visual documentation
```
Comment (jcheng5) [inline R/pr.R:501]:
```suggestion
#'     # turn off visual documentation
```

**Comment by jcheng5** [comment]
LGTM!
